model/library.py

```diff
@@ -292,6 +298,36 @@ def enabled_facets_setting(self, group_name):
         key = self.ENABLED_FACETS_KEY_PREFIX + group_name
         return self.setting(key)
 
+    @property
+    def has_root_lanes(self):
+        """Does this library have any lanes that act as the root
+        lane for a certain patron type?
+
+        :return: A boolean
+        """
+
+        # NOTE: Although this fact is derived from the Lanes, not the
+        # Library, the result is stored in the Library object for
+        # performance reasons.
+        #
+        # This makes it important to clear the cache of Library
+        # objects whenever the Lane configuration changes. Otherwise a
+        # library that went from not having root lanes, to having them
+        # (or vice versa) might not see the change take effect without
+        # a server restart.
+        value = Library._has_root_lane_cache.get(self.id, None)
+        if value is None:
+            from ..lane import Lane
+            _db = Session.object_session(self)
+            root_lanes = _db.query(Lane).filter(
+                Lane.library==self
+            ).filter(
+                Lane.root_for_patron_type!=None
```

Comment:
[PEP 8 Style Guide](https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_none.html) recommends to use `is` to compare something to `None`

Reply:
That's true, but this isn't really performing a comparison; it's creating a SQLAlchemy `BinaryExpression` object that gets turned into SQL like `Lane.root_for_patron_type is not NULL`. So using != is correct in this case.